sympy/vector/functions.py

```diff
@@ -474,3 +474,56 @@ def _path(from_object, to_object):
         from_path.append(other_path[i])
         i -= 1
     return index, from_path
+
+def orthogonalize(*vlist, **kwargs):
+    """
+    Takes a list of independent vectors and orthogonalizes them
+    using the Gram - Schmidt process. Returns a list of
+    orthogonal or orthonormal vectors.
+
+    Parameters
+    ==========
+
+    vlist : list of independent vectors to be made orthogonal.
+
+    orthonormal : Optional parameter
+    Set to True if the the vectors returned should be orthonormal.
+    Default: False
+
+    Examples
+    ========
+
+    >>> from sympy.vector.coordsysrect import CoordSysCartesian
+    >>> from sympy.vector.vector import Vector, BaseVector
+    >>> from sympy.vector.functions import orthogonalize
+    >>> C = CoordSysCartesian('C')
+    >>> i, j, k = C.base_vectors()
+    >>> v1 = i + 2*j
+    >>> v2 = 2*i + 3*j
+    >>> orthogonalize(v1, v2)
+    [C.i + 2*C.j, 2/5*C.i + (-1/5)*C.j]
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Gram-Schmidt_process
+
+    """
+    orthonormal = kwargs.get('orthonormal', False)
+
+    if not all(isinstance(vec, Vector) for vec in vlist):
+        raise TypeError('Each element must be of Type Vector')
+
+    ortho_vlist = []
+    for i in range(len(vlist)):
+        term = vlist[i]
+        for j in range(i):
+            term -= ortho_vlist[j].vec_project(vlist[i])
+        if term == Vector.zero:
```

Comment:
I didn't think about the process, but when comparing to zero in SymPy you have to make sure that Vector.zero is guaranteed to be returned. In many cases an expression that is equivalent to zero could be returned but it hasn't been reduced to zero because it may need to be simplified. If you are sure that Vector.zero is guaranteed to be returned here, then this is fine. If not, we may need to rethink this comparison. See #10279 for other info.